watson-developer-cloud/discovery-components
# fix: add document id to virtual scroll key

#### What do these changes do/fix?
Contributes to https://github.ibm.com/Watson-Discovery/disco-issue-tracker/issues/9712.

Fixes an issue where the naming of keys does not change when switching documents. This was causing styling issues, as the size of components would not be updated until the size of the window was adjusted.

Will require an update to the tooling repo, since this adds a new prop to the CIDocumentContent component (`documentId`).


#### How do you test/verify these changes?
I updated an existing storybook story based on the example in the connected issue (see above). Running the story for CIDocumentContent, do the following:
1. Change the knob for Document to ShortDocument. See that the styling looks normal.
2. Change the knob for Document to LongDocument. See that the segment holding the text looks like this:

  <img width="809" alt="Screen Shot 2022-04-27 at 9 53 08 AM" src="https://user-images.githubusercontent.com/11284492/165534677-80a26841-62a9-4e3c-b9c1-d4313353c6bf.png">

  and not this:

  <img width="671" alt="Screen Shot 2022-04-27 at 9 53 54 AM" src="https://user-images.githubusercontent.com/11284492/165534823-3e7120ca-db24-4f81-bb83-b245906ad338.png">
#### Have you documented your changes (if necessary)?

#### Are there any breaking changes included in this pull request?
